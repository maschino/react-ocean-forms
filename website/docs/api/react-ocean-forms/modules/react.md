---
id: "react"
title: "react"
sidebar_label: "react"
---

Copyright (c) 2018-present, Umweltbundesamt GmbH

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.

## Index

### Type aliases

* [PropsOf](react.md#propsof)

## Type aliases

###  PropsOf

Æ¬ **PropsOf**: *T extends function ? P : T extends object ? P : never*

Acquire the props for a Component {T}
