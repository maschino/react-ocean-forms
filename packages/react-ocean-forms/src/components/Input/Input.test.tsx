import React from 'react';

import { render } from '@testing-library/react';

import { Input } from './Input';
import { Form } from '../Form';

describe('<Input />', () => {
  it('should render without crashing', () => {
    const { asFragment } = render(
      <Form>
        <Input name="unitInput" label="unitInput" />
      </Form>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should only display the value if plaintext is set', () => {
    const { getByText, queryByLabelText } = render(
      <Form plaintext>
        <Input name="unitInput" label="unitInput" value="mock value" />
      </Form>
    );

    expect(getByText('mock value')).toBeVisible();
    expect(queryByLabelText('unitInput')).toBeNull();
  });

  it('should properly pass the type prop', () => {
    const { getByLabelText } = render(
      <Form>
        <Input name="unitInput" label="unitInput" type="number" />
      </Form>
    );

    expect(getByLabelText('unitInput')).toBeVisible();
    expect(getByLabelText('unitInput')).toHaveAttribute('type', 'number');
  });
});
