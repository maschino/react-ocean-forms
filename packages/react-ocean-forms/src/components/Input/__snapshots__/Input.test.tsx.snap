// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render without crashing 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <div
      class="field-group"
    >
      <label
        class="text-right"
        for="unitInput"
      >
        unitInput
      </label>
      <div
        class="input-container"
      >
        <input
          id="unitInput"
          name="unitInput"
          type="text"
          value=""
        />
      </div>
    </div>
  </form>
</DocumentFragment>
`;
